---
title: "MYH7B exon skipped ratio"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(DESeq2)
library(tximport)
source("util/_plotting.R")
```

```{r import_counts, message=FALSE, warning=FALSE, results='hide'}
# Read in gencode annotations
gtf <- rtracklayer::import("util/Rattus_norvegicus.Rnor_6.0.100.gtf")

# Renaming for compatibility with tximport
# TODO: Do we still need to do this?
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_id")] <-
  "geneID"
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_name")] <-
  "geneName"
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_type")] <-
  "geneBiotype"
gtf@elementMetadata$id <- gtf@elementMetadata$geneID

genes <- gtf[which(gtf$type == "gene")]
gtfdf <- gtf %>% as.data.frame()
g2s <- as.data.frame(gtf@elementMetadata@listData) %>% 
  dplyr::select(geneID, geneName) %>% 
  distinct()
tx2gene <- gtfdf %>%
  mutate(TXNAME = transcript_id, GENEID = geneID) %>%
  dplyr::select(TXNAME, GENEID)
```

```{r}
samples <- read.csv("samplesheet.csv")

samples$Condition <- sapply(samples$SampleType, function(x) {
  gsub('[[:digit:]]+', '', x)
})


files <- file.path("results/salmon", samples$SampleID, "quant.sf")
names(files) <- samples$SampleID
txisalmon <- tximport(files, type = "salmon", tx2gene = tx2gene, txOut = TRUE)
```

```{r}

tpm <- txisalmon$abundance %>%
  as.data.frame() %>%
  rownames_to_column("transcript_id")

write_csv(tpm, "results/transcript_tpm.csv")
```

```{r}
myh7b_gene_id <- g2s[grep("Myh7b", g2s$geneName),"geneID"]
myh7b_transcript_ids <- unique(tx2gene[grep(myh7b_gene_id, tx2gene$GENEID), "TXNAME"])
exon_skipped_transcript <- "ENSRNOT00000025859_exonskipped"
non_exon_skipped <- "ENSRNOT00000025859"


myh7b_tpm <- tpm[tpm$transcript_id %in% myh7b_transcript_ids, ] %>%
  pivot_longer(cols = 2:ncol(.), names_to = "run_accession", values_to = "tpm") %>%
  merge(samples)
```

```{r}
# let's make a plot!
g <- ggplot()
g + geom_point(data = myh7b_tpm, aes(x = transcript_id, y = tpm))


myh7b_tpm <- tpm[tpm$transcript_id %in% c(exon_skipped_transcript, non_exon_skipped), ] %>%
  pivot_longer(cols = 2:ncol(.), 
               names_to = "run_accession", 
               values_to = "tpm") %>%
  merge(samples)

myh7b_tpm$transcript_id <- factor(myh7b_tpm$transcript_id, levels = c("ENSRNOT00000025859_exonskipped", "ENSRNOT00000025859"),
                                        labels = c("Exon-skipped MYH7B", "MYH7B"))

library(ggbeeswarm)
g <- ggplot()
g + geom_boxplot(data = myh7b_tpm, aes(x = transcript_id, y = tpm)) + 
  theme_paperwhite() + geom_jitter(color = "#a8404c") + 
  ggtitle("Rat: MYH7B vs lncMYH7B") + 
  xlab("") +
  ylab("TPM")
ggsave("rat_myh7b_vs_lncmyh7b.pdf", height = 5, width = 2)
```




