---
title: "MYH7B exon skipped ratio"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(DESeq2)
library(tximport)
source("util/_plot_theme.R")
```

```{r import_counts, message=FALSE, warning=FALSE, results='hide'}
# Read in gencode annotations
gtf <- rtracklayer::import("util/gencode.v33.annotation.gtf")

# Renaming for compatibility with tximport
# TODO: Do we still need to do this?
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_id")] <-
  "geneID"
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_name")] <-
  "geneName"
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_type")] <-
  "geneBiotype"
gtf@elementMetadata$id <- gtf@elementMetadata$geneID

genes <- gtf[which(gtf$type == "gene")]
gtfdf <- gtf %>% as.data.frame()
g2s <- as.data.frame(gtf@elementMetadata@listData) %>% 
  dplyr::select(geneID, geneName) %>% 
  distinct()
tx2gene <- gtfdf %>%
  mutate(TXNAME = transcript_id, GENEID = geneID) %>%
  dplyr::select(TXNAME, GENEID)
```

```{r}
samples <- read.csv("sample_sheet.csv")
rownames(samples) <- samples$SampleID

files <- file.path("results/salmon", samples$SampleID, "quant.sf")
names(files) <- samples$SampleId
file.exists(files)
txisalmon <- tximport(files, type = "salmon", tx2gene = tx2gene, txOut = TRUE)
```

```{r}

tpm <- txisalmon$abundance %>%
  as.data.frame() %>%
  rownames_to_column("transcript_id")

write_csv(tpm, "results/transcript_tpm.csv")
```

```{r}
myh7b_gene_id <- g2s[grep("MYH7B", g2s$geneName),"geneID"]
myh7b_transcript_ids <- unique(tx2gene[grep(myh7b_gene_id, tx2gene$GENEID), "TXNAME"])
exon_skipped_transcript <- "ENST00000618182.5_exon_8_skipped"
non_exon_skipped <- "ENST00000618182.5"


myh7b_tpm <- tpm[tpm$transcript_id %in% myh7b_transcript_ids, ] %>%
  pivot_longer(cols = 2:ncol(.), names_to = "run_accession", values_to = "tpm") %>%
  merge(samples)
```

```{r}
# let's make a plot!
g <- ggplot(myh7b_tpm, aes(x = transcript_id, y = tpm))
g + geom_point()


myh7b_tpm <- tpm[tpm$transcript_id %in% c(exon_skipped_transcript, non_exon_skipped), ] %>%
  pivot_longer(cols = 2:ncol(.), 
               names_to = "run_accession", 
               values_to = "tpm") %>%
  merge(samples)

transcript7b_tpm <- filter(myh7b_tpm, transcript_id=="ENST00000618182.5")
lnctranscript7b_tpm <- filter(myh7b_tpm, transcript_id=="ENST00000618182.5_exon_8_skipped")


transcript7b_tpm$transcript_id <- factor(transcript7b_tpm$transcript_id, levels = "ENST00000618182.5",
                                        labels = "MYH7B")
lnctranscript7b_tpm$transcript_id <- factor(lnctranscript7b_tpm$transcript_id, levels = "ENST00000618182.5_exon_8_skipped",
                                        labels = "Exon-skipped MYH7B")

library(ggbeeswarm)
g <- ggplot()
g + geom_boxplot(data = transcript7b_tpm, aes(x = transcript_id, y = tpm)) + geom_boxplot(data = lnctranscript7b_tpm, aes(x = transcript_id, y = tpm)) + theme_paperwhite() + geom_jitter(color = "#a8404c") +
  ggtitle("Human Hearts: MYH7B vs lncMYH7B") + xlab("") + ylab("TPM")
ggsave("figures/humanheart_myh7b_vs_lncmyh7b.pdf", height = 5, width = 2)


```




